mhkit/dolfyn/velocity.py

```diff
@@ -1037,104 +770,33 @@ def power_spectral_density(self, veldat,
             fs = 2*np.pi*fs
             freq_units = 'rad/s'
             units = 'm^2/s/rad'
-            f_key = 'omega'
         else:
             freq_units = 'Hz'
             units = 'm^2/s^2/Hz'
-            f_key = 'f'
 
         # Spectra, if input is full velocity or a single array
         if len(veldat.shape) == 2:
-            assert veldat.shape[0] == 3, "Function can only handle 1D or 3D arrays"
+            assert veldat.shape[0] == 3, "Function can only handle 1D or 3D arrays." \
+                " If ADCP data, please select a specific depth bin."
 
             out = np.empty(self._outshape_fft(veldat[:3].shape),
                            dtype=np.float32)
             for idx in range(3):
-                out[idx] = self._psd_ndarray(veldat[idx], fs=fs, noise=noise[idx],
-                                             window=window, n_bin=n_bin,
-                                             n_pad=n_pad, n_fft=n_fft, step=step)
-            coords = {'S': ['Sxx', 'Syy', 'Szz'], time_str: time, f_key: freq}
-            dims = ['S', time_str, f_key]
+                out[idx] = self._psd_base(veldat[idx], fs=fs, noise=noise[idx],
+                                          window=window, n_bin=n_bin,
+                                          n_pad=n_pad, n_fft=n_fft, step=step)
+            coords = {'S': ['Sxx', 'Syy', 'Szz'], time_str: time, 'freq': freq}
+            dims = ['S', time_str, 'freq']
         else:
-            out = self._psd_ndarray(veldat, fs=fs, noise=noise[0], window=window,
-                                    n_bin=n_bin, n_pad=n_pad, n_fft=n_fft, step=step)
-            coords = {time_str: time, f_key: freq}
-            dims = [time_str, f_key]
-
-        da = xr.DataArray(out,
-                          name='psd',
-                          coords=coords,
-                          dims=dims,
-                          attrs={'units': units, 'n_fft': n_fft})
-        da[f_key].attrs['units'] = freq_units
-
-        return da
-
-    def cross_spectral_density(self, veldat,
```

Comment:
Was this function moved/renamed, or removed entirely?

Reply:
Moved this to ADV/turbulence.py for the same reason above